generate/generator/generate.go

```diff
@@ -565,6 +576,8 @@ func (variant DockerfileVariant) dockerBaseImage() string {
 		} else {
 			return fmt.Sprintf("ubuntu:%s", variant.ubuntuVersion())
 		}
+        case ProductEdgeServer:
+                return fmt.Sprintf("ubuntu:%s", variant.ubuntuVersion())
```

Comment:
The current implementation of `ubuntuVersion()` only supports Server (and doesn't do that very well either). The fact that SGW is also using ubuntuVersion() is a latent bug; once SGW hits 4.x, it's going to suddenly start making the Dockerfile be based on ubuntu:14.04. We should tidy that up and make it explicitly decide the version based on product.

Also, right now for Edge Server and SGW (and even Server) it's just selecting ubuntu:20.04. That's nearly EOL. We need to update that soon anyway. Do we know what Ubuntu version(s) Edge Server and SGW are certified for?

Reply:
Edge server is Ubuntu 22.04+

Reply:
ubuntuVersion() was changed a while back to support SGW and columnar. 
 https://github.com/couchbase/docker/commit/e7d7f2ea11a3d3643d2052f9c4ec98083d1ce9b9
 it is also modified to support couchbase-edge-server in this PR.

Reply:
My bad, I was looking at an out-of-date copy of this repo. Looks good now.